### PR TITLE
Feature/import error log

### PR DESCRIPTION
**Pull request by cloedu87**
- [x] catching error from validation change step
- [x] pushing them to csv stream
- [x] writing a file attachment and update import in async process to bypass transaction lock-in
- [x] adjust tests
- [x] refactoring
- [x] write specific test with huge amount of failing records

### CODE REVIEW

**Comment by cloedu87** [comment]
a test with a huge file which generates a lot of errors to check performance of the solution is still missing. but I will add this before merging - pinky promise 😄 - so feel free to start with the review already

File: lib/data_aggregator/records/import/import.ex [hunk 1]
```diff
@@ -270,6 +282,7 @@ defmodule DataAggregator.Records.Import do
     define :set_imported
     define :set_failed
     define :destroy
+    define :update_error_log, action: :update_error_log, args: [:error_log]
```
Comment (hwuethrich) [inline lib/data_aggregator/records/import/import.ex]: `action: ...` is redundant, I think, if they have the same name

File: priv/repo/migrations/20240604074043_fix_typo_in_attribute_name.exs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  """
+
+  use Ecto.Migration
+
+  def up do
+    rename table(:records), :eve_cover_toal_in_percentage, to: :eve_cover_total_in_percentage
+
+    alter table(:records) do
+      modify :eve_cover_total_in_percentage, :float
+    end
```
Comment (hwuethrich) [inline priv/repo/migrations/20240604074043_fix_typo_in_attribute_name.exs:15]: It this required?

File: lib/data_aggregator/records/import/changes/validate_rows.ex [hunk 3]
```diff
@@ -45,25 +45,35 @@ defmodule DataAggregator.Records.Import.Changes.ValidateRows do
...
+          {:cont, changeset}
+
+        {counts, errors}, changeset ->
+          write_error_log_file(error_log_file, errors)
+
+          changeset = report_progress(changeset, counts)
           {:cont, changeset}
       end)
 
+    upload_error_log_file!(path, import)
```
Comment (hwuethrich) [inline lib/data_aggregator/records/import/changes/validate_rows.ex:75]: Do you clean up the file somewhere? We currently use https://github.com/CargoSense/briefly for temporary files in Haystack.

**Review by hwuethrich — APPROVED** [review]

File: priv/repo/migrations/20240604074043_fix_typo_in_attribute_name.exs (hunk 2 above)
Comment (cloedu87) [reply priv/repo/migrations/20240604074043_fix_typo_in_attribute_name.exs:15]: it's a typo and as we use the name of the attributes in the frontend - at least for now - its more nice. there is anyway no production data yet, so no troubles migrating it on our staging system

File: lib/data_aggregator/records/import/changes/validate_rows.ex (hunk 3 above)
Comment (cloedu87) [reply lib/data_aggregator/records/import/changes/validate_rows.ex:75]: For now I just delete it - like `File.rm!(path)` - after I uploaded the file to s3, but I will address this in a global way in https://github.com/zebbra/data_aggregator/issues/301

**Pull request merged**